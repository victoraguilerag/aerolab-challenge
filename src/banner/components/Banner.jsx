import React from 'react'
import PropTypes from 'prop-types'
import image from '../../assets/header-x1.png'
import image2x from '../../assets/header-x2.png'

function Banner (props) {
  const { section } = props
  return (
    <div className="Banner" style={styles.banner}>
      <img src={image} srcSet={`${image2x} 2x`} style={styles.image} alt="banner background" />
      <h1 style={styles.section}>{section}</h1>
    </div>
  )
}

Banner.propTypes = {
  section: PropTypes.string.isRequired
}

const styles = {
  'banner': {
    'userSelect': 'none',
    'position': 'relative',
    'marginBottom': '48px'
  },
  'image': {
    'width': '100%'
  },
  'section': {
    'position': 'absolute',
    'bottom': '10.5%',
    'left': '11%',
    'fontSize': '64px',
    'color': '#ffffff',
    'textAlign': 'left'
  }
}

export default Banner
