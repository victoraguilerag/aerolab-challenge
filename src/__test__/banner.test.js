import React from 'react'
import Banner from '../banner/components/Banner.jsx'
import renderer from 'react-test-renderer'

test('Banner renders', () => {
  const component = renderer.create(
    <Banner section="Electronics" />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
