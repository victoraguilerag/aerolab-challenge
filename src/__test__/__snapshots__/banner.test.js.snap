// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders 1`] = `
<div
  className="Banner"
  style={
    Object {
      "marginBottom": "48px",
      "position": "relative",
      "userSelect": "none",
    }
  }
>
  <img
    alt="banner background"
    src={Object {}}
    srcSet="[object Object] 2x"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <h1
    style={
      Object {
        "bottom": "10.5%",
        "color": "#ffffff",
        "fontSize": "64px",
        "left": "11%",
        "position": "absolute",
        "textAlign": "left",
      }
    }
  >
    Electronics
  </h1>
</div>
`;
